src/aot.s

```diff
@@ -0,0 +1,69 @@
+.text
+
+
+.global _aot_trampoline
+_aot_trampoline:
+    // x0 <- fn_ptr: extern "C" fn()
+    // x1 <- args_ptr: *const u8
+    // x2 <- args_len: usize
+
+    //
+    // Check zero arguments.
+    //
+    cmp     x2,     0                       // If there are no arguments,
+    beq     3f                              // skip argument processing completely.
+
+    //
+    // Preparation.
+    //
+    add     x1,     x1,     x2,     lsl 3   // Move the pointer to the end (past last element).
+
+    str     x19,    [sp, #-8]!              // Push the x19 register (we'll use it to store sp).
+    mov     x19,    sp                      // Store the original stack pointer for later.
```

Comment:
I think we need to align the stack to 16 bytes. Maybe that's why we get `SIGBUS` for unaligned access.

Reply:
The 16-byte requirement is for x86. I'm not aware of ARM having that requirement.

Reply:
Maybe I'm misreading here, but I think Apple also puts that requirement: https://developer.apple.com/documentation/xcode/writing-arm64-code-for-apple-platforms#Pass-arguments-to-functions-correctly

Reply:
Oh you're right. I'm pretty sure I did search that link for alignment stuff but it seems I missed it or searched a different page. Thanks.